Dockerfile

```diff
@@ -3,74 +3,148 @@
-    zlib1g-dev \
-    zlib1g \
-    pip \
-    locales \
-    openssh-server \
-    dirmngr \
-    munge \
-    libmunge2 \
-    libmunge-dev \
-    libssl-dev \
-    libdbus-1-dev
+RUN apt-get update && \
+    apt-get --no-install-recommends -y install make \
+        automake \
+        autoconf \
+        build-essential \
+        bzip2 \
+        ca-certificates \
+        debianutils \
+        dirmngr \
+        g++ \
+        gcc \
+        git \
+        gpg \
+        gpg-agent \
+        libcurl4 \
+        libglib2.0-dev \
+        libgtk2.0-dev \
+        libmariadbd-dev \
+        libpam0g-dev \
+        libtool \
+        libncurses-dev \
+        libgdm1 \
+        libmunge2 \
+        libmunge-dev \
+        libssl-dev \
+        libdbus-1-dev \
+        mariadb-client \
+        mariadb-server \
+        munge \
+        wget \
+        zlib1g \
+        zlib1g-dev && \
+    update-ca-certificates
+
+RUN git clone -b ${SLURM_TAG} --single-branch --depth=1 https://github.com/SchedMD/slurm.git && \
+    cd slurm && \
+    ./configure \
+        --enable-debug \
+        --prefix=/usr \
+        --sysconfdir=/etc/slurm \
+        --with-mysql_config=/usr/bin \
+        --libdir=/usr/lib64 && \
+    make install && \
+    install -D -m644 etc/cgroup.conf.example /etc/slurm/cgroup.conf.example && \
+    install -D -m644 etc/slurm.conf.example /etc/slurm/slurm.conf.example && \
+    install -D -m644 etc/slurmdbd.conf.example /etc/slurm/slurmdbd.conf.example && \
+    install -D -m644 contribs/slurm_completion_help/slurm_completion.sh /etc/profile.d/slurm_completion.sh
+
+# ---------------------------------------------------------------------------------------------------------------------
+# Slurm runtime layer
+# NOTE: Without multi-stage builds, the final image has about 1.86 GB, with it has 1.08 GB (~ -41%).
 
-ARG SLURM_TAG
+FROM debian:stable-slim
 
-RUN git clone -b ${SLURM_TAG} --single-branch --depth=1 https://github.com/SchedMD/slurm.git \
-    && cd slurm \
-    && ./configure --enable-debug --prefix=/usr --sysconfdir=/etc/slurm \
-        --with-mysql_config=/usr/bin  --libdir=/usr/lib64 \
-    && make install \
-    && install -D -m644 etc/cgroup.conf.example /etc/slurm/cgroup.conf.example \
-    && install -D -m644 etc/slurm.conf.example /etc/slurm/slurm.conf.example \
-    && install -D -m644 etc/slurmdbd.conf.example /etc/slurm/slurmdbd.conf.example \
-    && install -D -m644 contribs/slurm_completion_help/slurm_completion.sh /etc/profile.d/slurm_completion.sh \
-    && cd .. \
-    && rm -rf slurm
+LABEL org.opencontainers.image.source="https://github.com/manuel-g-castro/slurm-cluster-openssh-docker/" \
+      org.opencontainers.image.title="slurm-cluster-openssh-docker" \
+      org.opencontainers.image.description="Slurm Docker cluster on Debian Slim with an OpenSSH server" 
 
+ENV LANG=en_US.UTF-8
+ENV LANGUAGE=en_US:en
+ENV LC_ALL=en_US.UTF-8
 # manuel: set the username for the run script
 ENV USERNAME=root
```

Comment:
I normally try to group things, or have some sort of organisation so it's easier to change/find things... in this file there are stages/containers, with LABEL, ARGS, ENVs, then a bunch of RUN/COPY/etc.

The dependencies were sorted alphabetically, so it's easier to check if something is needed.

Settings in Slurm were sorted with same logic.